osutil/io.go

```diff
@@ -78,6 +82,12 @@ func AtomicWriteFile(filename string, data []byte, perm os.FileMode, flags Atomi
 		return err
 	}
 
+	if uid > -1 && gid > -1 {
+		if err := fd.Chown(uid, gid); err != nil {
+			return err
+		}
+	}
```

Comment:
} else if uid > -1 || gid > -1 {
        return fmt.Errorf("internal error: AtomicWriteFileChown needs none or both of uid and gid set")
}